tests/Fixtures/Factories/PostFactory.php

```diff
@@ -22,6 +42,24 @@ public static function class(): string
         return Post::class;
     }
 
+    /**
+     * @return Proxy<Post>
+     * This method is here to test an edge case with the phpstan extension
+     */
+    public static function staticMethodCallWithSelf(): Proxy
+    {
+        return self::findOrCreate([]);
+    }
+
+    /**
+     * @return FactoryCollection<Proxy<Post>>
+     * This method is here to test an edge case with the phpstan extension
+     */
+    public function methodUsingThis(): FactoryCollection
+    {
+        return $this->sequence([]);
+    }
```

Comment:
this was easier to put these method here but I can create a factory specific for static analysis